### PR TITLE
Add others section to upstream servers pie chart

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

There are queries which are neither sent upstream nor are they cached or blocked. These are incorrectly ignored in the upstream server pie chart leading to a mismatch of the blocked percentage compared to the value reported at the top. Even when both are correct (their difference is caused by different reference frames), this is confusing. To compensate, we add a new type `other` which summarizes all these queries, causing the two reference frames to be identical.

### CODE REVIEW

**Review by yubiuser — COMMENTED** [review]
I think we need to add something here:
https://github.com/pi-hole/FTL/blob/7387e81981760c835148b7f5f61302b91ee83e7d/src/api/api.c#L723-L726

**Comment by yubiuser** [comment]
If adding "-3" here: I promised **Promo** https://github.com/pi-hole/AdminLTE/pull/1594

**Comment by DL6ER** [comment]
This shouldn't make any difference, the order is not changed. `Other` will use the color the first upstream had before, the first upstream gets the color for the former seconds, etc.

**Comment by rdwebdesign** [comment]
@yubiuser 

The code you highlighted is just to ensure that "BL" will be the first item (with index -2) and the second item will be cache (with index -1).

The rest remains in their original positions (from 0 forward).

**Review by yubiuser — APPROVED** [review]

File: src/api/api.c [hunk 1]
```diff
@@ -525,34 +525,45 @@ void getUpstreamDestinations(const char *client_message, const int *sock)
...
-		if(i == -2)
+		if(i == -3)
 		{
 			// Blocked queries (local lists)
 			ip = "blocklist";
 			name = ip;
 
-			if(totalqueries > 0)
+			if(counters->queries > 0)
 				// Whats the percentage of locked queries on the total amount of queries?
```
Comment (yubiuser) [inline src/api/api.c:545]:
```suggestion
				// Whats the percentage of blocked queries on the total amount of queries?
```

**Review by yubiuser — APPROVED** [review]

**Pull request merged**